spec/features/users/sign_in_spec.rb

```diff
@@ -131,14 +131,25 @@
         expect(page).to_not have_content(t('forms.buttons.submit.continue'))
 
         fill_in_credentials_and_click_sign_in(user.email, user.password)
-
         expect(page).to have_content t('errors.invalid_authenticity_token')
 
         fill_in_credentials_and_click_sign_in(user.email, user.password)
-
         expect(current_path).to eq user_two_factor_authentication_path
       end
     end
+
+    it 'displays the session timeout modal, does not allow the user to submit', js: true do
+      allow(Devise).to receive(:timeout_in).and_return(0)
+      user = create(:user)
+      visit root_path
+      fill_in 'Email', with: user.email
+      fill_in 'Password', with: user.password
+
+      expect(page).to have_css('#session-expired-msg')
+      expect(page).to have_css('[type=submit][disabled]')
+      page.execute_script("document.getElementById('new_user').submit();")
+      expect(page).not_to have_content t('errors.invalid_authenticity_token')
```

Comment:
This returns true with or without the js - is it necessary or is it sufficient to just make sure the submit button is disabled?